### DISCUSSION TITLE
alt_text missing

### DISCUSSION THREAD

**Discussion by rola93** [tweepy/tweepy#1571] [Questions] [answered]
Hello, I'm using tweepy 3.10 and I need to check whether a tweet contains images or not, and in case that images are included, if alt_text is given too.

However, I found some tweets does contain image and alt_text but doesn't I can't get it with the API.

[Consider this twet](https://twitter.com/IDATHAuy/status/1383120257565474816): I get it with the following line:

```python
status = api.get_status(tweet_id, include_ext_alt_text=True, include_entities=True)
```

But `status` object does not contain `extended_entities` attribute, so I can't go to `status.extended_entities['media'][0]['ext_alt_text']` to get alt_text. Notice that this work for some other cases like [this](https://twitter.com/amendaroUY/status/1382138812571193345).

Any ideas? my full code is [here](https://github.com/rola93/altBot/blob/1b44ae424a952c75e8ba29ad98e2f2d8d306f9fc/altBot_main.py#L271) if you want to check it, any help is appreciated.

thanks!

**Comment by Harmon758** [comment, marked as answer]
You'll want to use extended `tweet_mode`.
See [the section in the documentation about extended Tweets](https://tweepy.readthedocs.io/en/latest/extended_tweets.html).

**Reply by rola93** [reply]
Woow, this was the fastest reply I've ever got! thanks so much, it worked!

Just a small suggestion, maybe this should be said in the [`tweepy.API.get_status`](https://docs.tweepy.org/en/latest/api.html#tweepy.API.get_status) docs.

Once again, thanks for your answer!

**Reply by Harmon758** [reply]
No problem.
`tweet_mode` is applicable to every Twitter API v1.1 method / endpoint that retrieves Tweets, not just specifically to `API.get_status`, and I'm not sure adding a note to each method would be ideal.

**Reply by rola93** [reply]
As a non-expert user (I started last week) it took me hours to understand what was going on, even because some tweets did come back with `extended_entities` on it while others didn't. in fact, this specific method contains an specific parameter for this, `include_ext_alt_text` which not always work and that made me think there was something else wrong (the tweet, tweeter or my account) but all I needed was this extended argument.

In summary, may be just a link to the doc you provided here is enough.

Once again, great work: the fact that in one week I could write my first bot is only possible with your hard work, so thanks you again!
